handshake_ix.go

```diff
@@ -73,17 +73,28 @@ func ixHandshakeStage0(f *Interface, vpnIp uint32, hostinfo *HostInfo) {
 func ixHandshakeStage1(f *Interface, addr *udpAddr, hostinfo *HostInfo, packet []byte, h *Header) bool {
 	var ip uint32
 	if h.RemoteIndex == 0 {
-		ci := f.newConnectionState(false, noise.HandshakeIX, []byte{}, 0)
-		// Mark packet 1 as seen so it doesn't show up as missed
-		ci.window.Update(1)
-
+		ci := f.newConnectionState(false, noise.HandshakeIX, f.PSK, 0)
 		msg, _, _, err := ci.H.ReadMessage(nil, packet[HeaderLen:])
-		if err != nil {
+		if err != nil && len(f.AltPSKs) > 0 {
 			l.WithError(err).WithField("udpAddr", addr).
 				WithField("handshake", m{"stage": 1, "style": "ix_psk0"}).Error("Failed to call noise.ReadMessage")
-			return true
+			for _, key := range f.AltPSKs {
+				ci = f.newConnectionState(false, noise.HandshakeIX, key, 0)
+
+				msg, _, _, err = ci.H.ReadMessage(nil, packet[HeaderLen:])
+				if err != nil {
+					//l.WithError(err).WithField("udpAddr", addr).
+					//	WithField("handshake", m{"stage": 1, "style": "ix_psk0"}).Error("Failed to call noise.ReadMessage")
+				} else {
+					break
```

Comment:
nit: Slap in a comment that explains this picks the first parseable (non-erorr) alternate PSK in the order provided?